storage/innobase/handler/handler0alter.cc

```diff
@@ -8433,6 +8433,178 @@ innobase_update_foreign_cache(
 	DBUG_RETURN(err);
 }
 
+/** Changes SYS_COLUMNS.PRTYPE for one column.
+@param[in,out]	trx	transaction
+@param[in]	tableid	table ID as in SYS_TABLES
+@param[in]	name	table name
+@param[in]	prtype	new precise type
+@return		boolean flag
+@retval	true	on failure
+@retval false	on success */
+static
+bool
+innobase_change_field_versioning_try(
+	trx_t* trx,
+	const ulint tableid,
+	const char* name,
+	ulint prtype)
+{
+	DBUG_ENTER("innobase_change_field_versioning_try");
+
+	pars_info_t* info = pars_info_create();
+
+	pars_info_add_int4_literal(info, "prtype", prtype);
+	pars_info_add_ull_literal(info,"tableid", tableid);
+	pars_info_add_str_literal(info, "name", name);
+
+	dberr_t error = que_eval_sql(info,
+				     "PROCEDURE CHANGE_COLUMN_MTYPE () IS\n"
+				     "BEGIN\n"
+				     "UPDATE SYS_COLUMNS SET PRTYPE=:prtype\n"
+				     "WHERE TABLE_ID=:tableid AND NAME=:name;\n"
+				     "END;\n",
+				     false, trx);
+
+	trx->error_state = DB_SUCCESS;
```

Comment:
Why do we blindly reset `error_state` here?

Reply:
Because other parts of code does the same, f.ex. [here](https://github.com/MariaDB/server/blob/b534a7b89ef39cd75bfe09524be044bc16ea2f48/storage/innobase/handler/handler0alter.cc#L3760) and [here](https://github.com/MariaDB/server/blob/b534a7b89ef39cd75bfe09524be044bc16ea2f48/storage/innobase/handler/handler0alter.cc#L7989)

I've decided it's a preferred way to handle errors in this code. But now I can see a bit different code examples like [this one](https://github.com/MariaDB/server/blob/b534a7b89ef39cd75bfe09524be044bc16ea2f48/storage/innobase/handler/handler0alter.cc#L7687)

I'll try to understand that a bit better.

Reply:
`que_eval_sql` sets `trx->error_state` and returns it. So I think it's ok to reset it and return saved value.